protocol/development-workflow/README.md

```diff
@@ -0,0 +1,125 @@
+Development Workflow
+====================
+
+Setting Up the Release Branch
+-----------------------------
+
+As a team, determine the scope of a release. The release can contain one
+Pivotal Tracker story, or it can combine multiple stories that cannot be
+deployed independently of one another.
+
+Create a release branch off of `master`.
+
+    git checkout master
+    git pull
+    git checkout -b <release-name>
+
+Push the release branch to GitHub.
+
+    git push -u origin <release-name>
+
+Developing for the Release
+--------------------------
+
+Create a development branch off of the release branch.
+
+    git checkout <release-name>
+    git pull
+    git checkout -b <development-branch-name>
+
+Commit changes.
+
+Rebase over the release branch.
+
+    git fetch origin
+    git rebase origin/<release-name>
+
+Push the development branch to GitHub.
+
+    git push -u origin <development-branch-name>
+
+Submit a pull request to merge into the release branch.
+
+Conduct Code Review.
+
+Merge the development branch into the release branch.
+
+Follow the above process as many times as necessary to complete all
+components of the release in development branches.
+
+Conducting QA
+-------------
+
+Get the most recent state of the release branch.
+
+    git checkout <release-name>
+    git pull
+
+Create a staging branch off of `master`.
+
+    git checkout master
+    git pull
+    git checkout -b stage-<release-name>
+
+Merge the release branch into the release's staging branch.
+
+    git merge <release-name>
+
+Deliver the staging branch to the staging deployment.
+
+    git push staging master
+
+Wait for the deployment to finish.
+
+Conduct QA through the staging app front-end.
+
+Rejecting
+---------
+
+Rollback the database migrations of this release on staging.
+
+Reset the staging remote.
+
+    git checkout master
+    git push staging master
+
+Delete the release's staging branch.
+
+    git branch -D stage-<release-name>
+
+Indicate in Pivotal Tracker that the release was rejected.
+
+Accepting and Deploying
```

Comment:
@aaronagray: One of the major goals of this change is to _decouple_ merges from staging. In other words, we would no longer be doing any automatic deployments to staging. The reasoning behind this is that, instead, we want to be able to couple successful QA to deployment.

I believe that @KrishnaKulkarni is suggesting that we can find some mechanism within the GitHub UI to do one or both of these:

1. (When all components of a release are delivered) prepare a staging branch--that is, a fork of `master` with the release branch merged in--and deploy it to staging. NOTE: The current protocol calls for the staging branch name to indicate the release it contains. If the release branch is `my-release`, this branch would be `stage-my-release`.
2. (After item (1) followed by successful manual QA), integrate the staging branch into GitHub's `master`, re-stage (to keep staging up to date), and deploy.